src/main/java/uk/gov/hmcts/divorce/solicitor/service/CcdAccessService.java

```diff
@@ -107,6 +107,26 @@ public void unlinkUserFromApplication(Long caseId, String userToRemoveId) {
         log.info("Successfully unlinked applicant from case Id {} ", caseId);
     }
 
+    @Retryable(value = {FeignException.class, RuntimeException.class})
+    public void unlinkUserFromCase(Long caseId, String userToRemoveId) {
+        User caseworkerUser = idamService.retrieveSystemUpdateUserDetails();
+
+        final var creatorAssignmentRole = getCaseAssignmentUserRole(caseId, null, CREATOR.getRole(), userToRemoveId);
+        final var app2AssignmentRole = getCaseAssignmentUserRole(caseId, null, APPLICANT_2.getRole(), userToRemoveId);
+
+        final var caseAssignmentUserRolesReq = CaseAssignmentUserRolesRequest.builder()
+            .caseAssignmentUserRolesWithOrganisation(List.of(creatorAssignmentRole, app2AssignmentRole))
+            .build();
+
+        caseAssignmentApi.removeCaseUserRoles(
+            caseworkerUser.getAuthToken(),
+            authTokenGenerator.generate(),
+            caseAssignmentUserRolesReq
+        );
+
+        log.info("Successfully unlinked user from case (id: {}) ", caseId);
```

Comment:
Is it worth stating which roles are being unlinked? If we somehow had a situation where both creatorAssignmentRole and app2AssignmentRole were null, no actual unlinking would have happened.